src/o365/spo/commands/tenant/tenant-recyclebinitem-restore.ts

```diff
@@ -0,0 +1,149 @@
+import request from '../../../../request';
+import commands from '../../commands';
+import SpoCommand from '../../../base/SpoCommand';
+import GlobalOptions from '../../../../GlobalOptions';
+import { CommandOption, CommandCancel } from '../../../../Command';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  url: string;
+  wait: boolean;
+}
+
+interface RestSpoOperation {
+  HasTimedout: boolean;
+  IsComplete: boolean;
+  PollingInterval: number;
+}
+
+class SpoTenantRecycleBinItemRestoreCommand extends SpoCommand {
+  private timeout?: NodeJS.Timer;
+
+  public get name(): string {
+    return commands.TENANT_RECYCLEBINITEM_RESTORE;
+  }
+
+  public get description(): string {
+    return 'Restores the specified deleted Site Collection from Site Collection Recycle Bin';
```

Comment:
Shouldn't this be: _...from tenant recycle bin_?

Reply:
Indeed. I wasn't sure which term choose between "Site Collection" and "Tenant": you find both everywhere. But as we're working onlyn with online version only, it's better to refer to "Tenant".